addons/mail/models/mail_activity_type.py

```diff
@@ -125,3 +125,13 @@ def _inverse_triggered_next_type_id(self):
                 activity_type.chaining_type = 'trigger'
             else:
                 activity_type.chaining_type = 'suggest'
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_todo(self):
+        if self.env.ref('mail.mail_activity_data_todo') in self:
+            raise UserError("The 'to-do' activity type is used to create reminders from the top bar menu and the command palette. Consequently, it cannot be archived or deleted.")
```

Comment:
It should translated.
```suggestion
            raise UserError(_("The 'To-Do' activity type is used to create reminders from the top bar menu and the command palette. Consequently, it cannot be archived or deleted."))
```